app/src/lib/stores/git-store.ts

```diff
@@ -967,6 +971,32 @@ export class GitStore extends BaseStore {
     throw new Error(`Could not load commit: '${sha}'`)
   }
 
+  /**
+   * Refreshes the list of GitHub Desktop created stash entries for the repository
+   */
+  public async loadStashEntries(): Promise<void> {
+    if (!enableStashing()) {
+      return
+    }
+
+    const map = new Map<string, IStashEntry>()
+    const entries = await getDesktopStashEntries(this.repository)
+
+    for (const entry of entries) {
+      if (!map.has(entry.branchName)) {
```

Comment:
I'm curious about this check - can you add a comment here about why we skip over subsequent entries for a given branch after the first one?

Reply:
I think this is skipping over all entries that aren't already a part of the map. I think @outofambit can give a better answer than I though.

Reply:
hmm it looks like this skips subsequent entries. i _think_ this is because we only want the first matching one we find in the stash list. i can add a comment for this.

Reply:
added in 273dd01a995c5a1c92c2f5ace9711944829b1901